src/main/scala/com/databricks/spark/sql/perf/tpcds/TPCDS_2_4_Queries.scala

```diff
@@ -0,0 +1,3839 @@
+/*
+ * Copyright 2015 Databricks Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.databricks.spark.sql.perf.tpcds
+
+import com.databricks.spark.sql.perf.{Benchmark, ExecutionMode, Query}
+
+/**
+ * This implements the official TPCDS v1.4 queries with only cosmetic modifications
+ * (noted for each query).
+ * Don't modify this except for these kind of modifications.
+ */
+trait Tpcds_2_4_Queries extends Benchmark {
+
+  import ExecutionMode._
+
+  // Queries the TPCDS 2.3 queries using the qualifcations values in the templates.
+  val tpcds2_4Queries = Seq(
+      ("q1", """
```

Comment:
nit: How about we move each query into a separate file for better readability? Here's how we did this in Spark: https://github.com/apache/spark/tree/master/sql/core/src/test/resources/tpcds/